apps/sms/test/marionette/notification_test.js

```diff
@@ -225,7 +226,63 @@ marionette('Message notification tests', function() {
 
       openNotification();
 
-      assertMessagesIsInCorrectState();
+      assertMessagesIsInCorrectStateFromMessage(smsMessage);
+    });
+  });
+
+  suite('Display a conversation from a notification', function() {
+    var messagesStorage;
+    var thread1, thread2;
+
+    setup(function() {
+      thread1 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['999']
+      });
+      thread2 = ThreadGenerator.generate({
+        numberOfMessages: 50,
+        participants: ['888']
+      });
+
+      messagesStorage = [ thread1, thread2 ];
+    });
+
+    test('Clicking a notification while a conversation is loading', function() {
+      messagesApp.launch();
+      storage.setMessagesStorage(messagesStorage);
+      // Set empty contacts store.
+      storage.setContactsStorage();
+
+      var inbox = new InboxView(client);
+      var conversation = inbox.goToConversation(
+        thread1.id, { shouldWait: false }
+      );
+
+      // Receive 'notification' system message and fill up the storage.
+      messagesApp.sendSystemMessage('notification', {
+        clicked: true,
+        data: { threadId: thread2.id, id: thread2.messages[0].id }
+      });
+
+      conversation.accessors.waitToAppear();
```

Comment:
if I wait neither earlier nor here, I'll have the infamous "crash but can't stackwalk" message :/ Which doesn't help much. From my tests, it seems that we need at least that the window is unhidden (but we don't really care that it's been transformed)